src/polydatum/dal.py

```diff
@@ -65,19 +107,15 @@ def __getitem__(self, path):
 
             dal['myservice.get'](my_id)
         """
+        path_segments = []
         paths = path.split('.')
-        p = paths.pop(0)
-        if p == 'dal':
-            p = paths.pop(0)
-        loc = self._services[p]
-        while 1:
-            try:
-                p = paths.pop(0)
-            except IndexError:
-                break
-            else:
-                loc = getattr(loc, p)
-        return loc
+        for p in paths:
+            if p != 'dal':
+                path_segments.append(PathSegment(name=p))
+        ctx = self._data_manager.require_active_context()
+        return dal_resolver(
```

Comment:
This should return the same thing `__getattr__` does. This does not need a context until `__call__` is called. Doing `dal["foo.bar"]()` is the same as doing `dal.foo.bar()`.

- [ ] Return a request builder -- not a resolved method/service.

Reply:
Great catch, my mistake. Notably we saved this change for last and I confidently told Harini "we can trust the other tests to cover this, since we're not intending to actually change any behavior." So I suspect we found a gap. Cool!

- [ ] we need to add a test, or figure out why the existing one(s) passed

Reply:
I think returning the real method works because...it's the real method. It just won't go through any middleware. So the test should probably check that middleware was ran.

Reply:
We just skimmed through the existing tests, I didn't see a test that covers item access like this at all - did I miss one? I don't see any specific tests for it, and skimmed the other existing tests I didn't see any implicit test of that kind of access either.